R/IDateTime.R

```diff
@@ -83,6 +83,21 @@ round.IDate = function (x, digits=c("weeks", "months", "quarters", "years"), ...
           years = ISOdate(year(x), 1L, 1L)))
 }
 
+trunc.IDate = function (x, units = c("weeks", "months", "quarters", "years"), start.on.monday = TRUE, ...) {
```

Comment:
I would make this more flexible than `cut.Date()`, though I'm not sure the exact signature that's most helpful.

Is this too flexible?

```r
week_start = "Mon"

offset = if (is.numeric(week_start)) offset else switch(week_start,
  M = , Mon = ,  Monday = 4L,
  Tu = , Tue = ,  Tuesday = 5L,
  W = , Wed = , Wednesday = 6L,
  Th = , Thu = , Thursday = 0L,
  F = , Fri = , Friday = 1L,
  Sa = , Sat = , Saturday = 2L,
  Su = , Sun = , Sunday = 3L)
```